tests/xml_parser_test.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/auth/private/xml_parser.h>
+#include <aws/testing/aws_test_harness.h>
+
+const char *root_with_text = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><rootNode>TestBody</rootNode>";
```

Comment:
Should there be a test with some CDATA in it, just for completeness sake?

Reply:
haha, it would likely fail. I haven't implemented CDATA support and would prefer not to since this is private and only for STS responses at the moment.